src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -255,7 +266,7 @@ public JsonPropertyInfo GetProperty(ReadOnlySpan<byte> propertyName, ref ReadSta
                 // First try sorted lookup.
                 int propertyIndex = frame.PropertyIndex;
 
-                // This .Length is consistent no matter what json data intialized _propertyRefsSorted.
+                // Cache the .Length in case it changes by another thread. It may become longer but never shorter (thus IndexOutOfRangeException avoided).
```

Comment:
There should be a comment in UpdateSortedPropertyCache detailing how important it is that the ordering of existing items in _propertyRefsSorted be maintained when it grows.

Reply:
With the second commit, the ordering is not important.